### ISSUE TITLE
[Tools] Content about Parcel is deprecated and troublesome

### ISSUE THREAD

**Issue by devusexu** [mdn/content#28080]
### MDN URL

https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Package_management

### What specific section or headline is this issue about?

Having fun with Parcel & Building our code for production

### What information was incorrect, unhelpful, or incomplete?

I list the three commands I fail to follow along
### npm install parcel-bundler 
This package has been [deprecated](https://www.npmjs.com/package/parcel-bundler), should just install [parcel](https://www.npmjs.com/package/parcel)
Note that I would love to use the deprecated version but it can't even be installed anymore

I know the article would mention the version issue later but it should be fixed in this early part unless the author want  the audience to find out themselves

### parcel index.html
TL;DR --- Parcel failed to auto install 
* parcel didn't auto install the date-fns package but reporting build failed instead
    *  @parcel/transformer-js: Browser scripts cannot have imports or exports.

* Following the suggestion I added type="module" to the script tag. The problem was not solved while other errors occurred 
   * @parcel/core: Failed to resolve 'date-fns' from './index.js'
   * @parcel/resolver-default: Cannot find module 'date-fns'

* To find out whether parcel can auto install packages, I tried add a file named "style.sass" and include it in a link tag in index.html, parcel did actually install sass for me with 
  ```
  "devDependencies": {
    "@parcel/transformer-sass": "^2.9.3"
  }
  ```
* I thought there was something wrong with date-fns, but then I tried another popular package [dayjs](https://www.npmjs.com/package/dayjs), it failed with the same error messages
   * @parcel/core: Failed to resolve 'dayjs' from './index.js'
   * @parcel/resolver-default: Cannot find module 'dayjs'

While I fail to solve, I think I should at least post this issue.

### parcel build index.html
error: 
* @parcel/namer-default: Target "main" declares an output file path of "index.js" which does
	not match the compiled bundle type "html".

	solution: [remove the "main" field in package.json](https://stackoverflow.com/questions/68884914/where-should-the-path-from-main-in-package-json-point-to)

### Conclusion
I think this part should be updated to some extent since it's hard to follow along, and I can not even figure out the solution.

My environment: 
Windows 11 / Ubuntu 22.04.2 LTS run on WSL
node: v18.17.0 / v20.4.0
parcel: 2.9.3

	

### What did you expect to see?

Commands that can be followed along like all other tutorials
To be specific: 
1. Viable parcel version
2. Successful package auto installation
3. Successfully parcel build with default command and npm init settings

### Do you have anything more you want to share?

![Pasted image 20230720174909](https://github.com/mdn/content/assets/138584931/c8e7d4a4-3efa-4604-8023-ed4620ee21e0)
![Pasted image 20230720175107](https://github.com/mdn/content/assets/138584931/0575af18-8ae6-4aa9-b033-d99b7a8111ee)
![Pasted image 20230720175132](https://github.com/mdn/content/assets/138584931/5a55dabc-8e82-496e-acb1-68cdb246bef1)
![Pasted image 20230720175452](https://github.com/mdn/content/assets/138584931/481a9605-7cf7-4ac8-8916-6435d096b95c)

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/learn/tools_and_testing/understanding_client-side_tools/package_management`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Understanding_client-side_tools/Package_management
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/tools_and_testing/understanding_client-side_tools/package_management/index.md
* Last commit: https://github.com/mdn/content/commit/acfe8c9f1f4145f77653a2bc64a9744b001358dc
* Document last modified: 2023-07-07T07:19:19.000Z

</details>
